src/field.rs

```diff
@@ -190,6 +196,72 @@ pub trait FieldElement:
         }
         Ok(vec)
     }
+
+    /// Encode `input` as `bits`-bit vector of elements of `Self` if it's small enough
+    /// to be represented with that many bits.
+    ///
+    /// # Arguments
+    ///
+    /// * `input` - The field element to encode
+    /// * `bits` - The number of bits to use for the encoding
+    fn encode_into_bitvector_representation(
+        input: &Self::Integer,
+        bits: usize,
+    ) -> Result<Vec<Self>, FieldError> {
+        // Create a mutable copy of `input`.
+        // In each iteration of the following loop
+        // we take the least significant bit,
+        // and shift input to the right by one bit.
+        let mut i = *input;
+
+        let one = Self::Integer::from(Self::one());
+        let mut encoded = Vec::with_capacity(bits);
+        for _ in 0..bits {
+            let w = Self::from(i & one);
+            encoded.push(w);
+            i = i >> one;
+        }
+
+        // If `i` is still not zero, this means that
+        // it cannot be encoded by `bits` bits.
+        if i != Self::Integer::from(Self::zero()) {
+            return Err(FieldError::InputSizeMismatch);
+        }
+
+        Ok(encoded)
+    }
+
+    /// Decode the bitvector-represented value `input` into a simple representation as a single field element.
+    /// This function errors if `2^input.len() - 1` does not fit into the field `Self`.
+    fn decode_from_bitvector_representation(input: &[Self]) -> Result<Self, FieldError> {
+
+        Self::valid_integer_bitlength(input.len())?;
+
+        let mut decoded = Self::zero();
+        for (l, bit) in input.iter().enumerate() {
+            let w = Self::Integer::try_from(1 << l).map_err(|_| {FieldError::IntegerTryFrom})?;
+            decoded += Self::from(w) * *bit;
+        }
+        Ok(decoded)
+    }
+
+    /// Interpret `i` as [`Self::Integer`] if it's representable in that type and smaller than the field modulus.
+    fn valid_integer_try_from<N>(i: N) -> Result<Self::Integer, FieldError> where Self::Integer: TryFrom<N>{
+        let i_int = Self::Integer::try_from(i).map_err(|_| {FieldError::IntegerTryFrom})?;
+        if Self::modulus() <= i_int {
+            return Err(FieldError::ModulusOverflow);
+        }
+        Ok(i_int)
+    }
+
+    /// Check if the largest number representable with `bits` bits (i.e. 2^bits - 1) is representable in this field.
+    fn valid_integer_bitlength(bits: usize) -> Result<(), FieldError> {
```

Comment:
I think it would be better if this method just returned a `bool` indicating whether the `2^bits-1` is encodable and let the caller return an error if not.